psl/psl/src/lib.rs

```diff
@@ -49,6 +49,14 @@ pub fn parse_configuration_multi_file(
     psl_core::parse_configuration_multi_file(files, builtin_connectors::BUILTIN_CONNECTORS)
 }
 
+/// Parses and validates Prisma schemas, but skip analyzing everything except datasource and generator
+/// blocks. It never fails, but when the returned `Diagnostics` contains errors, it implies that the 
+/// `Configuration` content is partial.  
+/// Consumers may then decide  whether to convert `Diagnostics` into an error.
+pub fn infallible_parse_configuration(files: &[(String, SourceFile)]) -> (Files, Configuration, Diagnostics) {
```

Comment:
I keep getting confused by this being called infallible and thinking ah, yes, no issues at all, wait, why are we getting diagnostics, what happened to our infallible leader!?

Reply:
@SevInf, maybe `parse_config_best_effort` could better emphasise the role of this function?

Reply:
@jkomyno @Druue what would you say about `error_tolerant_parse_configuration`?